https://issues.redhat.com/browse/ACM-14268 (#6974)
open-cluster-management/rhacm-docs: business_continuity/backup_restore/backup_existing_hub.adoc

[#backup-existing-hub]

> = Using an existing hub as a restore hub 

Reviewer: Reminder to add this to the business_continuity/main.adoc file

> = Using an existing hub as a restore hub 

Reviewer: ```suggestion
= Using an existing hub cluster as a restore hub cluster
```

> If you use a restore hub as a cluster with resources that a user created before the restore operation, you must understand the following constraints: 

Reviewer: Let's add a short description before this. Something like, "When you use a restore hub cluster with resources that you created before running the restore operation, you can tag user-created resources on the restore hub cluster with the `velero.io/backup-name: backupName` label."

then go into the constraints.

* The existing managed cluster can be detached during the restore process.

> * The restore hub can have more resources than the backup hub.

Reviewer: ```suggestion
* The restore hub cluster can have more resources than the backup hub cluster.
```

[#existing-clusters-detached]
== Existing managed clusters can be detached during the restore process

If the hub you plan to use for the restore operation already manages some clusters, expect the following outcomes when you run a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`:  

> * If an earlier backup created the `ManagedCluster` resources, then they have the `velero.io/backup-name: backupName` label annotation set. Therefore, if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, the clusters are detached and the data is cleaned up on those on those managed clusters. This only occurs if these managed clusters resources are not part of the restored backup.

Reviewer: ```suggestion
* If an earlier backup created the `ManagedCluster` resources, then the `velero.io/backup-name: backupName` label annotation is set. Therefore, if you run a restore operation and use the `cleanupBeforeRestore=CleanupRestored` option, the clusters are detached and the data is cleaned up on the managed clusters. This only occurs if the backup managed clusters resources are not part of the restored backup.
```

> * If an earlier backup created the `ManagedCluster` resources, then they have the `velero.io/backup-name: backupName` label annotation set. Therefore, if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, the clusters are detached and the data is cleaned up on those on those managed clusters. This only occurs iff these managed clusters resources are not part of the restored backup.

Reviewer 2: typo : iff 
content is good !

> * If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, the user must run a restore operation and use the `cleanupBeforeRestore=CleanupRestored` option. Tese managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub, and this hub does not match the content of the backup hub.

Reviewer 2: Line 15 is still incorrect .. we changed the meaning .. what I meant to say: if the resource has the label **and** the user runs a restore operation using the CleanupRestored option, **then** the managed clusters are not cleaned up, as they should, because they don't have that label.

Please change line 15 with 

If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation **and** the user runs a restore operation **with** the `cleanupBeforeRestore` option set to `CleanupRestored`, **then** these managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub, and this hub does not match the content of the backup hub.

> * If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, and the user runs a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`, then these managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub, and this hub does not match the content of the backup hub.

Reviewer: Based on the contents of this section, it seems like you are describing how the resources are cleaned or not cleaned with the user has the parameter set. It seems like when you say detached, you are saying that the contents of the backup hub cluster and restore hub cluster do not match. 

Is it more accurate to rename the section title to "Outcomes for existing managed clusters during the restore process"?

Author: Thanks! I made the change. I changed the title to "Understanding.." to be consistent with how most of our docs start with a gerund. 

@birsanv please let me know if you agree with this title change.

> * If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, and the user runs a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`, then these managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub, and this hub does not match the content of the backup hub.

Reviewer: The following statement is long:

Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub, and this hub does not match the content of the backup hub.

What is "this hub"? Is it the restore hub cluster? 

```suggestion
* If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, and you run a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`, then the managed cluster resources are not cleaned up since they were not created by an earlier backup. Therefore, the managed clusters continue to be managed by this hub cluster, even though the clusters were not managed by the backup hub cluster, and this hub does not match the content of the backup hub cluster.
```

[#restore-hub-resources]

> == Restore hub can have more resources than the backup hub

Reviewer: Similar to my earlier comment, can it be renamed?

Author: I can make the change if you want. These are more possible outcomes, hence saying, "can have more..." 

If you really don't like it, I can think of something else :)

If your restore operation uses a hub with earlier created user data, then the restore operation results in a hub that does not have the exact content as the primary hub. The user data includes the following resources: `policy`, `application`, `credential`, or any custom resource defined by the user. A restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored` is not going to clean up these resources because they were created by the users and not created by an earlier backup. 

> If you intend to use an existing hub for a restore hub configuration and expect that the hub has the same content as the backup after the restore is completed, tag any resource that would be part of a hub backup with the `velero.io/backup-name: backupName` label annotation. Tagging the resources with the Velero annotation marks them as being generated by an earlier restore.

Reviewer 2: small change here : 
This : and expect that the hub has the same content as the backup 
Should be : and expect that the hub has the same content as the backup **hub**

[#prerequisite-resources]
=== Prerequisite 

> * On the restore hub, update the `DataProtectionApplication` resource and set the storage location to a new, temporary location. You do not want to create the backup used to tag the restore hub resources in the same storage location with the backup hub. Before you tag the resources, make sure you set the storage location to a different location than the backup hub.

Reviewer: The statement "Before you tag the resources, make sure you set the storage location to a different location than the backup hub." seems like another prereq

----

. Verify that the `BackupSchedule` has created the hub backup resources and their status is `Completed`.
. Pause the `BackupSchedule` schedule and stop generating backups by setting the property `paused=true` on the `schedule-acm-msa BackupSchedule`.
.. *Note:* You cannot run a restore operation on the hub cluster if the `BackupSchedule` is enabled, so you pause the `BackupSchedule` to run a restore operation.
. Run a restore operation by using the following yaml: 

+
[source,yaml]
----
kind: Restore
metadata:
name: restore-acm
namespace: open-cluster-management-backup
spec:
cleanupBeforeRestore: None
veleroManagedClustersBackupName: latest
veleroCredentialsBackupName: latest
veleroResourcesBackupName: latest
excludedResources: # exclude managed cluster resources
 - ManagedCluster
excludedNamespaces: # exclude managed cluster namespaces
  - managed-cls-1
  - managed-cls-2
  - managed-cls-n
----

After you complete these steps, the operator goes over each resource which is part of the hub backup and tags it with the `velero.io/backup-name: backupName` , where `backupName` is the name of the latest backup. 

[#understanding-resources]
=== Understanding the tagged resources 

> After tagging the resources, you can use the hub as a restore cluster. Update the `DataProtectionApplication` storage location to point to the storage location of the backup hub so you can run a restore operation using the hub backup resources. All the restore hub resources are tagged now with the `velero.io/backup-name label`, so they show as being generated by an earlier restore. Therefore, any restore operation that you run on this hub cluster using the `cleanupBeforeRestore` option set to `CleanupRestored` cleans up the data that is not part of the restored backup.

Reviewer: It seems like this should be moved earlier in the Tagging resources section, rather than be a new subsection

Author: Valentina and I did talk about putting this earlier, for instance, part of the intro for === Tagging resources

However, this content is only applicable _after_ users tag the resource. Given the Doc's team's recent discussions of using more headings to organize content, I made it its own section. 

However, if you think it's best, I can remove the heading, === Understanding the tagged resources 

That way users see this content right after competing the task/procedure. Thoughts?

[#backup-existing-hub]
= Using an existing hub as a restore hub 

If you use a restore hub as a cluster with resources that a user created before the restore operation, you must understand the following constraints: 

* The existing managed cluster can be detached during the restore process.
* The restore hub can have more resources than the backup hub.

[#existing-clusters-detached]
== Existing managed clusters can be detached during the restore process

If the hub you plan to use for the restore operation already manages some clusters, expect the following outcomes when you run a restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored`:  

> * If an earlier backup created the `ManagedCluster` resources, then they have the `velero.io/backup-name: backupName` label annotation set. Therefore, if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, then these `ManagedCluster` resources are not part of the restored backup. The clusters are detached and the data is cleaned up on those managed clusters. 

Reviewer 2: This is incorrect
Change

Therefore, if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, then these `ManagedCluster` resources are not part of the restored backup. The clusters are detached and the data is cleaned up on those managed clusters.

With:

Therefore, if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, then the clusters are detached and the data is cleaned up on those  on those managed clusters, if these managed clusters resources are not part of the restored backup.

> * When the  `velero.io/backup-name: backupName` label annotation exists, the managed clusters were created by an earlier backup and are no longer part of the restored backup. As a result, the managed clusters are detached, and the data is cleaned up on those managed clusters if the user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option .

Reviewer 2: Remove line 15, it says the same thing as line 14

> * If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub. Therefore, this hub does not match the content of the backup hub. 

Reviewer 2: Change line 15 with : 

If the `ManagedCluster` resource does not have the `velero.io/backup-name: backupName` label annotation, user runs a restore operation and uses the `cleanupBeforeRestore=CleanupRestored` option, these managed cluster resources are not cleaned up since they were not created by a previous backup. Therefore, these managed clusters continue to be managed by this hub, even though these clusters were not managed by the backup hub; this hub does not match the content of the backup hub.

[#restore-hub-resources]
== Restore hub can have more resources than the backup hub

> If your restore operation uses a hub with earlier created user data, then the restore operation results in a hub that does not have the exact content as the primary hub. The user data includes the following properties: `policy`, `application`, `credential`, or any custom resource defined by the user. A restore operation with the `cleanupBeforeRestore` option set to `CleanupRestored` is not going to clean up these resources because they were created by the users and not created by an earlier backup. 

Reviewer 2: This 
The user data includes the following properties

Should be
The user data includes the following **resources**

If you intend to use an existing hub for a restore hub configuration and expect that the hub has the same content as the backup after the restore is completed, tag any resource that would be part of a hub backup with the `velero.io/backup-name: backupName` label annotation. Tagging the resources with the Velero annotation marks them as being generated by an earlier restore.

[#prerequisite-resources]
=== Prerequisite 

> * Update the `DataProtectionApplication` resource and set the storage location to a new, temporary location. You do not want to have this backup in the same location with the backup hub, so make sure you set the storage location to a different location.

Reviewer 2: Change
Update the `DataProtectionApplication` resource and set the storage location to a new, temporary location. You do not want to have this backup in the same location with the backup hub, so make sure you set the storage location to a different location.

With ( I was trying to make this more clear: 1. you run on the restore hub, 2. we are talking about the backup generated by the Tagging resources section ):

**On the restore hub**, update the `DataProtectionApplication` resource and set the storage location to a new, temporary location. You do not want to create the backup **used to tag the restore hub resources** in the same **storage** location with the backup hub, **so before you run the next step** make sure you set the storage location to a different location than the backup hub.

[#tagging-resources]
=== Tagging resources 

> To tag resources with the `velero.io/backup-name: backupName` label, complete the following steps: 

Reviewer 2: Change  : 

 To tag resources with the `velero.io/backup-name: backupName` label, complete the following steps:

With 

To tag **user created resources on the restore hub** with the `velero.io/backup-name: backupName` label, **on the restore hub** complete the following steps:

.  Create a `BackupSchedule` resource to generate a backup that contains all these hub resources by using the following yaml:

+
[source,yaml]
----
apiVersion: cluster.open-cluster-management.io/v1beta1
kind: BackupSchedule
metadata:
 name: schedule-acm-msa
 namespace: open-cluster-management-backup
spec:
 veleroSchedule: 0 */1 * * *
 veleroTtl: 120h
 useManagedServiceAccount: true
 paused: false
----

> . Pause the `BackupSchedule` schedule and stop generating backups by setting the property `paused=true` on the `schedule-acm-msa BackupSchedule`.

Reviewer 2: Add this before Line 52 (ask the user to verify they got a hub backup ) : 

. Verify that the BackupSchedule has created the hub Backup resources and their status is Completed.

.. *Note:* You cannot run a restore operation on the hub cluster if the `BackupSchedule` is enabled, so you pause the `BackupSchedule` to run a restore operation.
. Run a restore operation by using the following yaml: 

+
[source,yaml]
----
kind: Restore
metadata:
name: restore-acm
namespace: open-cluster-management-backup
spec:
cleanupBeforeRestore: None
veleroManagedClustersBackupName: latest
veleroCredentialsBackupName: latest
veleroResourcesBackupName: latest
excludedResources: # exclude managed cluster resources
 - ManagedCluster
excludedNamespaces: # exclude managed cluster namespaces
  - managed-cls-1
  - managed-cls-2
  - managed-cls-n
----

After you complete these steps, the operator goes over each resource which is part of the hub backup and tags it with the `velero.io/backup-name: backupName` , where `backupName` is the name of the latest backup. 

[#understanding-resources]

> === Understanding the resources

Reviewer 2: I am not sure Understanding the resources is a proper title, what resources are we referring to ?..

Maybe : Understanding the procedure 
( assuming we are referring to this tagging of resources procedure )

> You can now use the hub as a restore cluster and update the `DataProtectionApplication` storage location to point to the storage location of the backup hub. Any restore operation that you set the `cleanupBeforeRestore` option to `CleanupRestored` cleans up data that is not part of the latest restored backup. All the hub backup resources are tagged with the `velero.io/backup-name` label, so they show as being generated by an earlier restore.

Reviewer 2: Please change line 81 with

You can now use the hub as a restore cluster. Update the `DataProtectionApplication` storage location to point to the storage location of the backup hub **so you can run a restore operation using the hub backup resources**.  All the restore hub resources are tagged now with the `velero.io/backup-name` label, so they show as being generated by an earlier restore. Therefore, any restore operation **that you run on this hub cluster** using the `cleanupBeforeRestore` option set to `CleanupRestored` cleans up the data that is not part of the restored backup.

> *Important:*

Reviewer 2: - Remove lines 85 and 86
- Move line 91 to be first on the **Important** section

* If your hub already manages other clusters, they get detached. If you run any restore operation where you use the `cleanupBeforeRestore` set to `CleanupRestored` and if these clusters are not part of the restored backup and they are tagged with the `velero.io/backup-name` label, then the managed cluster resources get removed. Therefore, use the restore resource to exclude any managed cluster resources so that you do not need to tag them with the `velero.io/backup-name` label.
* Do not run a restore operation where you use the `cleanupBeforeRestore` set to `CleanupRestored`. If the restore hub already has managed clusters, and if they were created by an earlier backup, they are tagged with the `velero.io/backup-name` label. If you use this option, the managed clusters get detached and data on the managed clusters get cleaned up. 

> * Having this constraint, you could use an existing hub for a restore operation in the following situations: 

Reviewer 2: Change
Having this constraint, you could use an existing hub for a restore operation in the following situations:

With
Having this constraint, you could use an existing **hub as a restore hub** in the following situations:
